node/sc/sub_bridge_handler.go

```diff
@@ -357,37 +373,45 @@ func (sbh *SubBridgeHandler) broadcastServiceChainReceiptRequest() {
 	}
 }
 
-func (sbh *SubBridgeHandler) blockAnchoringManager(block *types.Block) {
-	startBlkNum := sbh.GetNextAnchoringBlockNumber()
-
-	var successCnt, cnt, blkNum uint64
-	latestBlkNum := block.Number().Uint64()
+// updateTxCounts update txCounts to insert into anchoring tx. It skips first remnant txCounts.
+func (sbh *SubBridgeHandler) updateTxCounts(block *types.Block) {
+	if sbh.txCountsEnabledBlockNumber == 0 {
+		sbh.txCounts = 0 // reset for the next anchoring period
+		sbh.txCountsEnabledBlockNumber = block.NumberU64()
+		if sbh.chainTxPeriod > 1 {
+			remnant := block.NumberU64() % sbh.chainTxPeriod
+			if remnant < 2 {
+				sbh.txCountsEnabledBlockNumber += 1 - remnant
+			} else {
+				sbh.txCountsEnabledBlockNumber += (sbh.chainTxPeriod - remnant) + 1
+			}
+		}
+	}
 
-	// TODO-Klaytn-Servicechain remove this code or define right confirmation block count.
-	// To consider a non-absolute finality consensus like PoA (Clique), this code anchors past blocks.
-	const confirmCnt = uint64(10)
-	if latestBlkNum > confirmCnt {
-		latestBlkNum = latestBlkNum - confirmCnt
+	var startBlkNum uint64
+	if sbh.latestAnchoredBlockNumber == 0 {
+		startBlkNum = block.NumberU64()
 	} else {
-		return
+		startBlkNum = sbh.latestAnchoredBlockNumber + 1
```

Comment:
We need to change the variable name like below.
`latestAnchoredBlockNumber` -> `latestAddedBlockNumber`